src/test/scala/org/coroutines/async-await-tests.scala

```diff
@@ -309,25 +307,47 @@ class AsyncAwaitTest extends FunSuite with Matchers {
     assert(result == 103)
   }
 
-  // Source: https://git.io/vrFj3
+  /** Source: https://git.io/vrFj3
+   *  Modified so that there is no coroutine passed as a by-name parameter. For more
+   *  information, see https://git.io/vKkM5.
+   */
   test("nested await as bare expression") {
     val c = async(coroutine { () =>
-      await(Future(await(Future("")).isEmpty))
+      val emptyString = await(Future(""))
```

Comment:
`Future.apply` takes a by-name parameter, but coroutines can't be passed by-name. See #24.

I will add some unit tests to assert that this check happens.